### DISCUSSION TITLE
Zeek file extraction issue

### DISCUSSION THREAD

**Discussion by rohan0017** [Security-Onion-Solutions/securityonion#11616] [Unsupported Versions]
In my security onion distributed deployment, the zeek module is not capturing any files. Any help is appreciated. The files are not going into the /nsm/zeek/extracted/complete folder.

**Comment by robbiemarshall** [comment]
Are you running Strelka?  If so, it may be moving the extracted files out of the /nsm/zeek/extracted/complete folder to /nsm/strelka/processed/, so you can check that folder to see if the files are there.

**Reply by rohan0017** [reply]
I checked but the files are not coming there

https://github.com/Security-Onion-Solutions/securityonion/discussions/11616#discussioncomment-7389869

**Reply by rohan0017** [reply]
Is zeek by default not enabled when we first install Security Onion

**Reply by rohan0017** [reply]
![gitbug_screenshot](https://github.com/Security-Onion-Solutions/securityonion/assets/131326479/09a8471f-e1e0-4102-b56f-14caa366fffe)

**Reply by TOoSmOotH** [reply]
You are using https thus the connection is encrypted. Try grabbing something using http.

**Reply by rohan0017** [reply]
So zeek does not extract any files if I am using https. Does zeek extract files only from ftp, http, smtp etc?

**Reply by rohan0017** [reply]
I tried using http, then also zeek is not capturing the files.

![github2](https://github.com/Security-Onion-Solutions/securityonion/assets/131326479/12fc0f24-9d19-48cb-9260-ae461d0d27a2)

**Reply by dougburks** [reply]
Is Zeek logging metadata about the file transfer in its http.log, file.log, and conn.log?

Is there any packet loss? If so, Zeek will not extract files.

**Reply by rohan0017** [reply]
No, there is no packet loss this is my packet_loss.log content=> {"ts":1698469291.631522,"ts_delta":45309.525259017944,"peer":"zeek","gaps":0,"acks":0,"perce    nt_lost":0.0}

**Reply by rohan0017** [reply]
And there are no http.log and file.log log files in logs directory.

**Reply by dougburks** [reply]
Double-check the configuration of your tap or span port and the wiring that connects it to your sniffing interface. Then use tcpdump to verify that you see the HTTP traffic arriving at the sniffing interface.

**Comment by rohan0017** [comment]
How can I test whether the zeek is extracting the files or not(from traffic)?

**Reply by dougburks** [reply]
One option would be to use `curl` to download an EXE file over HTTP from a machine whose traffic is being monitored by your Security Onion deployment:
```
curl http://testmynids.org/exe/calc.exe -o /tmp/calc.exe
```

**Reply by rohan0017** [reply]
I tried it, still zeek is not extracting any files. It is not dropping any packets, I Double-checked the configuration of my span port and the wiring that connects it to my sniffing interface, everything looks fine.

**Reply by dougburks** [reply]
Does the Zeek http.log show that it saw your connection to testmynids.org?

**Reply by rohan0017** [reply]
I am able to see only these logs: 
/nsm/zeek/spool/logger# ls
broker.log  capture_loss.log  conn.log  dhcp.log  dns.log  ecat_arp_info.log  enip.log  notice.log  stats.log  stderr.log  stdout.log

http.log is missing

**Reply by rohan0017** [reply]
And in stderr.log zeek is logging the following errors: 

 /opt/zeek/share/zeekctl/scripts/run-zeek: line 61: ulimit: core file size: cannot modify limit: Operation not permitted
send-mail: /usr/sbin/sendmail not found
send-mail: /usr/sbin/sendmail not found
send-mail: /usr/sbin/sendmail not found
send-mail: /usr/sbin/sendmail not found

**Reply by dougburks** [reply]
It seems like your sensor may only be seeing broadcast traffic so there may still be a problem with your span configuration.

**Reply by rohan0017** [reply]
Ok I will check.
